test/media-groups.test.js

```diff
@@ -18,1208 +18,1426 @@ QUnit.module('MediaGroups', {
+        'main audio set when changing to group with playlist loader'
+      );
+      assert.notOk(
+        mainSegmentLoaderSetAudioCalls[3],
+        'main audio set to false when changing to group with playlist loader'
+      );
+      assert.equal(segmentLoaderSetAudioCalls.length, 3, 'audio set on track change');
+      assert.ok(segmentLoaderSetAudioCalls[2], 'audio enabled on track change');
+      assert.equal(
+        segmentLoaderResetCalls, 2,
+        'reset on track change'
+      );
+      assert.strictEqual(
+        mediaType.activePlaylistLoader, groups.main[1].playlistLoader,
+        'sets the correct active playlist loader'
+      );
+      assert.strictEqual(
+        segmentLoaderTrack, tracks.fr,
+        'set the correct track on the segment loader'
+      );
+    }
+  );
 
-    tracks.en.enabled = false;
-    tracks.fr.enabled = true;
-    mediaType.activePlaylistLoader = groups.main[2].playlistLoader;
+  const createMocker = ({calls = [], args = []} = {}) => {
+    const obj = {calls: {}, args: {}};
 
-    onGroupChanged();
+    calls.forEach(function(key) {
+      obj.calls[key] = 0;
+      obj[key] = () => {
+        obj.calls[key]++;
+      };
+    });
 
-    assert.equal(segmentLoaderPauseCalls, 4, 'loaders paused on group change');
-    assert.equal(
-      mainSegmentLoaderResetCalls, 1,
-      'no reset changing to group with playlist loader'
-    );
-    assert.equal(
-      segmentLoaderResyncCalls, 1,
-      'resync changing to group with playlist loader'
-    );
-    assert.strictEqual(
-      mediaType.activePlaylistLoader, groups.main[1].playlistLoader,
-      'sets the correct active playlist loader'
-    );
-  }
-);
-
-QUnit.test(
-  'onTrackChanged updates active playlist loader and resets segment loader',
-  function(assert) {
-    let mainSegmentLoaderResetCalls = 0;
-    const mainSegmentLoaderSetAudioCalls = [];
-    let segmentLoaderResetCalls = 0;
-    const segmentLoaderSetAudioCalls = [];
-    let segmentLoaderPauseCalls = 0;
-    let segmentLoaderTrack;
+    args.forEach(function(key) {
+      obj.args[key] = [];
+      obj[key] = (v) => {
+        obj.args[key].push(v);
+      };
+    });
+
+    return obj;
+  };
 
+  const mockSegmentLoader = () => createMocker({
+    calls: ['abort', 'pause', 'resetEverything'],
+    args: ['setAudio', 'track']
+  });
+  const mockPlaylistLoader = () => createMocker({
+    calls: ['pause', 'load'],
+    args: ['fastQualityChange_']
+  });
+
+  const mocksAreZero = (mocks, assert) => {
+    Object.keys(mocks).forEach(function(name) {
+      const mock = mocks[name];
+
+      Object.keys(mock.calls).forEach(function(key) {
+        assert.equal(mock.calls[key], 0, `${name} ${key} not called`);
+      });
+
+      Object.keys(mock.args).forEach(function(key) {
+        assert.equal(mock.args[key].length, 0, `${name} ${key} not called`);
+      });
+    });
+  };
+
+  QUnit.test('onTrackChanged with isMasterPlaylist', function(assert) {
```

Comment:
The previous `onTrackChanged` test was very brittle, so I didn't want to add anything else to it. I instead added the this test here that tests the new `isMasterPlaylist` functionality.